packages/strapi-plugin-users-permissions/controllers/User.js

```diff
@@ -70,19 +70,27 @@ module.exports = {
    */
 
   create: async (ctx) => {
-    if ((await strapi.store({
+    const advanced = await strapi.store({
       environment: '',
       type: 'plugin',
       name: 'users-permissions',
       key: 'advanced'
-    }).get()).unique_email && ctx.request.body.email) {
+    }).get();
+
+    if (advanced.unique_email && ctx.request.body.email) {
       const user = await strapi.query('user', 'users-permissions').findOne({ email: ctx.request.body.email });
 
       if (user) {
         return ctx.badRequest(null, ctx.request.admin ? [{ messages: [{ id: 'Auth.form.error.email.taken', field: ['email'] }] }] : 'Email is already taken.');
       }
     }
 
+    if (!ctx.request.body.role) {
+      const defaultRole = await strapi.query('role', 'users-permissions').findOne({ type: advanced.default_role }, []);
+
+      ctx.request.body.role = defaultRole._id || defaultRole.id;
```

Comment:
FYI, we need to check `_id || id` because Mongoose has a default getter on the `id` attribute (if the lean() method isn't enabled) (see http://mongoosejs.com/docs/guide.html#id)

Reply:
Yap ! Just copy and past it from an other file.